### PR TITLE
Clean up nvpair block processing

### PR DESCRIPTION
**Pull request by kgaillot**
This is prep work for moving the remaining libpe_rules code to libcrmcommon

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , here's a low-priority review when you get a chance. The last one for real

**Review by nrwahl2 — COMMENTED** [review]
nice

**Comment by nrwahl2** [comment]
> last one for real

Your call, you're almost there ;)

File: lib/pengine/rules.c [hunk 1]
```diff
@@ -121,7 +121,7 @@ populate_hash(xmlNode * nvpair_list, GHashTable * hash, gboolean overwrite, xmlN
          an_attr != NULL; an_attr = pcmk__xe_next(an_attr)) {
 
         if (pcmk__xe_is(an_attr, PCMK_XE_NVPAIR)) {
-            xmlNode *ref_nvpair = expand_idref(an_attr, top);
+            xmlNode *ref_nvpair = expand_idref(an_attr, NULL);
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/pengine/rules.c:128]: Yeah I'm fine with that

File: lib/pengine/rules.c [hunk 2]
```diff
@@ -85,10 +84,12 @@ sort_pairs(gconstpointer a, gconstpointer b)
         return -1;
     }
 
-    if (pcmk__str_eq(pair_a->name, pair_a->special_name, pcmk__str_casei)) {
+    if (pcmk__str_eq(pcmk__xe_id(pair_a->attr_set), pair_a->special_name,
+                     pcmk__str_none)) {
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/pengine/rules.c:85]: oops

File: lib/pengine/rules.c [hunk 3]
```diff
@@ -208,7 +209,7 @@ unpack_attr_set(gpointer data, gpointer user_data)
  */
 static GList *
 make_pairs(const xmlNode *xml_obj, const char *set_name,
-           const char *always_first, gboolean overwrite)
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/pengine/rules.c:204]: actually it was already unused! previously the function used sorted_set_t:overwrite and ignored the argument

File: lib/pengine/rules.c [hunk 4]
```diff
@@ -228,7 +228,6 @@ make_pairs(const xmlNode *xml_obj, const char *set_name,
             }
 
             pair = pcmk__assert_alloc(1, sizeof(sorted_set_t));
-            pair->special_name = always_first;
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/pengine/rules.c:225]: thx

**Comment by kgaillot** [comment]
updated per review

**Review by nrwahl2 — APPROVED** [review]

File: lib/pengine/rules.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/pengine/rules.c:204]: ah right

**Comment by nrwahl2** [comment]
retest this please

**Pull request merged**